pkg/golinters/whitespace.go

```diff
@@ -24,27 +21,58 @@ func NewWhitespace(settings *config.WhitespaceSettings) *goanalysis.Linter {
 	var wsSettings whitespace.Settings
 	if settings != nil {
 		wsSettings = whitespace.Settings{
+			Mode:      whitespace.RunningModeGolangCI,
 			MultiIf:   settings.MultiIf,
 			MultiFunc: settings.MultiFunc,
 		}
 	}
 
-	analyzer := &analysis.Analyzer{
-		Name: whitespaceName,
-		Doc:  goanalysis.TheOnlyanalyzerDoc,
-		Run:  goanalysis.DummyRun,
-	}
+	whitespaceAnalyzer := whitespace.NewAnalyzer(&wsSettings)
 
 	return goanalysis.NewLinter(
-		whitespaceName,
-		"Tool for detection of leading and trailing whitespace",
-		[]*analysis.Analyzer{analyzer},
+		whitespaceAnalyzer.Name,
+		whitespaceAnalyzer.Doc,
+		[]*analysis.Analyzer{whitespaceAnalyzer},
 		nil,
 	).WithContextSetter(func(lintCtx *linter.Context) {
-		analyzer.Run = func(pass *analysis.Pass) (any, error) {
-			issues, err := runWhitespace(lintCtx, pass, wsSettings)
-			if err != nil {
-				return nil, err
+		whitespaceAnalyzer.Run = func(pass *analysis.Pass) (any, error) {
+			whitespaceIssues := whitespace.Run(pass, &wsSettings)
+			issues := make([]goanalysis.Issue, len(whitespaceIssues))
+
+			for i, issue := range whitespaceIssues {
+				report := &result.Issue{
+					FromLinter: whitespaceAnalyzer.Name,
+					Pos:        pass.Fset.PositionFor(issue.Diagnostic, false),
+					Text:       issue.Message,
+				}
+
+				switch issue.MessageType {
+				case whitespace.MessageTypeRemove:
+					report.LineRange = &result.Range{
+						From: issue.LineNumber,
+						To:   issue.LineNumber,
+					}
+
+					report.Replacement = &result.Replacement{
+						NeedOnlyDelete: true,
+					}
+
+				case whitespace.MessageTypeAdd:
+					position := pass.Fset.PositionFor(issue.Diagnostic, false)
+					bracketLine, err := lintCtx.LineCache.GetLine(position.Filename, position.Line)
```

Comment:
@ldez Can you help me out here? I would like to get rid of this as well but I must be missing something obvious.

I was hoping I could do just like with `SuggestedFixes` and get the column of the statement and add a `"\n"` of length 1 but this doesn't work as I thought. I was trying to do something like this:

```go
report.Pos = pass.Fset.PositionFor(issue.FixStart, false)
report.Replacement = &result.Replacement{
	Inline: &result.InlineFix{
		StartCol:  0,
		Length:    1,
		NewString: "\n",
	},
}
```

but this makes the tests fail with:

```diff
Diff:
--- Expected
+++ Actual
@@ -50,3 +50,3 @@

-       fmt.Println("Hello world")
+fmt.Println("Hello world")
 }
@@ -57,3 +57,3 @@

-               fmt.Println("Hello multi-line world")
+       fmt.Println("Hello multi-line world")
        }
```

One fix seems to be to always add an extra tab to `NewString` (set it to `\n\t`) and still keep length of 1. This makes the tests pass and it also works for a deeper example like this one:

```go
	if true {
		if true {
			if true {
				if 1 == 1 &&
					2 == 2 {
					fmt.Println("Hello multi-line world")
				}
			}
		}
	}
```

but is it reasonable to rely on `\t` and that the file uses tabs for the fixer? Maybe the fixer will always convert spaces to tabs even if `gofmt` isn't enabled?

Reply:
Our internal fixes are really different than SuggestedFixes.
I don't remember exactly but there is a trick and a limitation, I will try it locally.

Reply:
from my memories, inline cannot work with line**s**, so `\n` will not work.

Reply:
Thanks, sounds good to me! I'll go with the `\n\t` fix then because that way we don't need to use the lint cache to read the file.